resources/js/components/chapter-sort.js

```diff
@@ -0,0 +1,189 @@
+     * Setup the given book container element to have sortable items.
+     * @param {Element} chapterContainer
+     */
+    setupChapterSortable(chapterContainer) {
+        const sortElems = [chapterContainer.querySelector('.sort-list')];
+        sortElems.push(...chapterContainer.querySelectorAll('.sortable-page-list'));
+
+        const bookGroupConfig = {
+            name: 'chapter',
+            pull: ['chapter', 'page'],
+            put: ['chapter', 'page'],
+        };
+
+        const chapterGroupConfig = {
+            name: 'chapter',
+            pull: ['chapter', 'page'],
+            put: function(toList, fromList, draggedElem) {
+                return draggedElem.getAttribute('data-type') === 'page';
+            }
+        };
+
+        for (let sortElem of sortElems) {
+            new Sortable(sortElem, {
+                group: sortElem.classList.contains('sort-list') ? bookGroupConfig : chapterGroupConfig,
+                animation: 150,
+                fallbackOnBody: true,
+                swapThreshold: 0.65,
+                onSort: this.updateMapInput.bind(this),
+                dragClass: 'bg-white',
+                ghostClass: 'primary-background-light',
+                multiDrag: true,
+                multiDragKey: 'CTRL',
+                selectedClass: 'sortable-selected',
+            });
+        }
+    }
+
+    /**
+     * Update the input with our sort data.
+     */
+    updateMapInput() {
+        const pageMap = this.buildEntityMap();
+        this.input.value = JSON.stringify(pageMap);
+    }
+
+    /**
+     * Build up a mapping of entities with their ordering and nesting.
+     * @returns {Array}
+     */
+    buildEntityMap() {
+        const entityMap = [];
+        const lists = this.elem.querySelectorAll('.sort-list');
+
+        for (let list of lists) {
+            const chapterId = list.closest('[data-type="chapter"]').getAttribute('data-id');
+            const directChildren = Array.from(list.children)
+                .filter(elem => elem.matches('[data-type="page"]'));
+            for (let i = 0; i < directChildren.length; i++) {
+                this.addChapterChildToMap(directChildren[i], i, chapterId, entityMap);
+            }
+        }
+
+        return entityMap;
+    }
+
+    /**
+     * Parse a sort item and add it to a data-map array.
+     * Parses sub0items if existing also.
+     * @param {Element} childElem
+     * @param {Number} index
+     * @param {Number} chapterId
+     * @param {Array} entityMap
+     */
+    addChapterChildToMap(childElem, index, chapterId, entityMap) {
+        const type = childElem.getAttribute('data-type');
+        const parentChapter = false;
+        const childId = childElem.getAttribute('data-id');
+
+        entityMap.push({
+            id: childId,
+            sort: index,
+            parentChapter: parentChapter,
+            type: type,
+            chapter: chapterId
+        });
+
+    }
+
+}
+
+export default ChapterSort; 
```

Comment:
This entire class shares a massive amount of logic with the `book-sort.js` file. Could this not instead be made into a generic `content-sort.js` file? I just don't want duplicate classess/functions to maintain if possible.